src/components/input-number/input-number.scss

```diff
@@ -137,15 +137,15 @@
 
 // position placeholder/value text in relation to icons
 
-:host([icon][scale="s"]) input {
+:host([icon]:not([icon=""])[scale="s"]) input {
```

Comment:
If the doc and type no longer allows boolean, I think we can drop `:not([icon=""])` and assume `icon` has a value.